src/h2_connection.erl

```diff
@@ -1000,7 +1012,58 @@ handle_event(_, {send_headers, StreamId, Headers, Opts},
         closed ->
             {keep_state, Conn}
     end;
+handle_event(_, {send_trailers, StreamId, Headers, _Opts},
+             #connection{
+                encode_context=EncodeContext,
+                streams = Streams,
+                socket = _Socket
+               }=Conn
+            ) ->
+    lager:debug("[~p] {send trailers, ~p, ~p}",
+                [Conn#connection.type, StreamId, Headers]),
+    %% StreamComplete = proplists:get_value(send_end_stream, Opts, false),
+
+    Stream = h2_stream_set:get(StreamId, Streams),
+    case h2_stream_set:type(Stream) of
+        active ->
+            {FramesToSend, NewContext} =
+                h2_frame_headers:to_frames(h2_stream_set:stream_id(Stream),
+                                           Headers,
+                                           EncodeContext,
+                                           (Conn#connection.peer_settings)#settings.max_frame_size,
+                                           true
+                                          ),
+            OldBody = h2_stream_set:queued_data(Stream),
+
+            NewS = h2_stream_set:update_trailers(FramesToSend, Stream),
+            NewBody = case is_binary(OldBody) of
+                          true -> OldBody;
+                          false -> <<>>
+                      end,
```

Comment:
in what situations would `OldBody` not be a binary and why replace it in that case?

Reply:
`queued_data` can be `undefined` or `done`. So if it is one of those we just make an empty body to send the trailers.

Reply:
Hm, this may not be needed.